crates/kv-store/src/iter.rs

```diff
@@ -0,0 +1,476 @@
+use std::{
+    cell::RefCell,
+    collections::{binary_heap::PeekMut, BinaryHeap},
+    fmt::Debug,
+    rc::Rc,
+};
+
+use bytes::Bytes;
+
+pub trait KvIterator: Debug {
+    fn next_key(&self) -> Bytes;
+    fn next_value(&self) -> Bytes;
+    fn find_next(&mut self);
+    fn is_next_valid(&self) -> bool;
+    fn prev_key(&self) -> Bytes;
+    fn prev_value(&self) -> Bytes;
+    fn find_prev(&mut self);
+    fn is_prev_valid(&self) -> bool;
+}
+
+#[derive(Debug)]
+struct HeapIterWrapper<T> {
+    pub idx: usize,
+    pub iter: Rc<RefCell<T>>,
+    pub f2b: bool,
```

Comment:
What does this mean?

Reply:
I use a `BinaryHeap` to get the *first* `KvIterator`, and these iters are wrapped by `Rc<RefCell<T>>` for sharing between *next* and *next_back*.   So `f2b` means `front to back`. If true, the heap will pop the iter with the smallest key, otherwise, the biggest one.